src/tool/xmeta/xmeta_emit.h

```diff
@@ -0,0 +1,91 @@
+
+    class xmeta_emit : public xlang_model_listener
+    {
+    public:
+        xmeta_emit(std::string const& assembly_name) 
+            : m_assembly_name(assembly_name)
+        { };
+
+        void initialize();
+        void uninitialize();
+        void save_to_file() const;
+		std::vector<uint8_t> save_to_memory() const;
+
+        void listen_namespace_model(std::shared_ptr<namespace_model> const& model) override;
+        void listen_class_model(std::shared_ptr<class_model> const& model) override;
+        void listen_struct_model(std::shared_ptr<struct_model> const& model) override;
+        void listen_interface_model(std::shared_ptr<interface_model> const& model) override;
+        void listen_enum_model(std::shared_ptr<enum_model> const& model) override;
+        void listen_delegate_model(delegate_model const& model) override;
+
+        mdTypeDef define_type_def(std::string const& name, DWORD const& type_flag, mdToken token_extend, mdToken token_implements[]);
+        void define_method(std::shared_ptr<method_model> const& model, mdTypeDef const& token_def);
+        void define_property(std::shared_ptr<property_model> const& model, mdTypeDef const& token_def);
+        void define_event(std::shared_ptr<event_model> const& model, mdTypeDef const& token_def);
+
+        void define_return(std::optional<type_ref> const& retun_type, mdTypeDef const& type_def, mdParamDef *token_return);
+        void define_parameters(formal_parameter_model const& model, mdMethodDef const& token_method, int parameter_index);
+
+    private:
+        std::string m_assembly_name;
+        winrt::com_ptr<IMetaDataDispenserEx> m_metadata_dispenser;
+        winrt::com_ptr<IMetaDataAssemblyEmit> m_metadata_assembly_emitter;
+        winrt::com_ptr<IMetaDataEmit2> m_metadata_emitter;
+        winrt::com_ptr<IMetaDataImport> m_metadata_import;
+        std::map<std::string, meta::reader::TypeRef> type_references;
+
+        meta::reader::Module m_module;
+        mdAssembly token_assembly;
+        mdAssemblyRef token_mscorlib;
+        mdTypeRef token_enum;
+        mdTypeRef token_value_type;
+        mdTypeRef token_delegate;
+
+        // A generic assembly metadata struct.
+        static constexpr ASSEMBLYMETADATA s_genericMetadata =
+        {
+            // usMajorVersion - Unspecified major version
+            0xFF,
+            // usMinorVersion - Unspecified minor version
+            0xFF,
+            // usRevisionNumber - Unspecified revision number
+            0xFF,
+            // usBuildNumber - Unspecified build number
+            0xFF,
+            // szLocale - locale indepedence
+            nullptr,
+            // cbLocale
+            0,
+            // rProcessor - Processor independence
+            nullptr,
+            // ulProcessor
+            0,
+            // rOS - OS independence
+            nullptr,
+            // ulOS
+            0,
+        };
+
+        // Windows specific methods
```

Comment:
How are these methods Windows specific?

Reply:
Whoops stray comment. The whole file is windows specific.